pkg/solidity-utils/contracts/math/BasePoolMath.sol

```diff
@@ -186,13 +188,20 @@ library BasePoolMath {
 
         // Calculate the new invariant with fees applied.
         // This invariant should be lower than the original one, so we don't need to check invariant ratio bounds again.
-        uint256 invariantWithFeesApplied = pool.computeInvariant(newBalances);
+        // Rounding down makes bptAmountOut go down (see comment below).
+        uint256 invariantWithFeesApplied = pool.computeInvariant(newBalances, Rounding.ROUND_DOWN);
 
         // Calculate the amount of BPT to mint. This is done by multiplying the
         // total supply with the ratio of the change in invariant.
         // Since we multiply and divide we don't need to use FP math.
-        // Round down since we're calculating BPT amount out. `invariantWithFeesApplied` calculated with `newBalances`
-        // rounded down, which also contributes to a lower `bptAmountOut`.
+        // Round down since we're calculating BPT amount out. This is the most important result of this function,
+        // equivalent to:
+        // `totalSupply * (invariantWithFeesApplied / currentInvariant - 1)`
+        // Then, to round `bptAmountOut` down we use `invariantWithFeesApplied` rounded down and `currentInvariant`
```

Comment:
Consider some spacing "//" lines around the "Then..." paragraph for better readability.